npm_and_yarn/lib/dependabot/npm_and_yarn/package_manager.rb

```diff
@@ -10,6 +10,7 @@ module NpmAndYarn
     ECOSYSTEM = "npm_and_yarn"
     MANIFEST_FILENAME = "package.json"
     LERNA_JSON_FILENAME = "lerna.json"
+    PACKAGE_MANAGER_VERSION_REGEX = /^\d+\.\d+\.\d+(-[a-zA-Z0-9.]+)?(\+[a-zA-Z0-9.]+)?$/
```

Comment:
I'm not great at reading Regex like this. If it seems worth your while, what do you think of naming the captures? I think that would make it 🤏 easier for me to parse. That said, I think this is doing what we need it to 😄

Reply:
Sorry about that. You mean using something like the following?

```
PACKAGE_MANAGER_VERSION_REGEX = /^(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)(?:-(?<pre_release>[a-zA-Z0-9.]+))?(?:\+(?<build>[a-zA-Z0-9.]+))?$
```

Reply:
Yep 😄 _I_ find that much easier to parse. What do you think of it?

Reply:
Change is made.